geomstats/geometry/invariant_metric.py

```diff
@@ -163,11 +166,352 @@ def inner_product_matrix(self, base_point=None):
 
         metric_mat = gs.einsum(
             '...ij,...jk->...ik',
-            inv_jacobian_transposed, self.inner_product_mat_at_identity)
+            inv_jacobian_transposed, self.metric_mat_at_identity)
         metric_mat = gs.einsum(
             '...ij,...jk->...ik', metric_mat, inv_jacobian)
         return metric_mat
 
+    def structure_constant(self, tangent_vec_a, tangent_vec_b, tangent_vec_c):
+        r"""Compute the structure constant of the metric.
+
+        For three tangent vectors :math: `x, y, z` at identity,
+        compute  :math: `<[x,y], z>`.
+
+        Parameters
+        ----------
+        tangent_vec_a : array-like, shape=[..., dim]
+            Tangent vector at identity.
+        tangent_vec_b : array-like, shape=[..., dim]
+            Tangent vector at identity.
+        tangent_vec_c : array-like, shape=[..., dim]
+            Tangent vector at identity.
+
+        Returns
+        -------
+        structure_constant : array-like, shape=[...,]
+        """
+        bracket = GeneralLinear.bracket(tangent_vec_a, tangent_vec_b)
+        return self.inner_product_at_identity(bracket, tangent_vec_c)
+
+    def adjoint_star(self, tangent_vec_a, tangent_vec_b):
```

Comment:
NIT: why not calling it `dual_adjoint`, thus refering to its name and not its notation?

Reply:
yes much better